RFS/src/main/java/com/rfs/cms/CmsClient.java

```diff
@@ -41,5 +42,46 @@ public interface CmsClient {
      * Updates the Metadata Migration entry in the CMS.  Returns an Optional; if the document was updated,
      * it will be the updated entry and empty otherwise.
      */
-    public Optional<CmsEntry.Metadata> updateMetadataEntry(CmsEntry.MetadataStatus status, String leaseExpiry, Integer numAttempts);
+    public Optional<CmsEntry.Metadata> updateMetadataEntry(CmsEntry.Metadata newEntry, CmsEntry.Metadata lastEntry);
+
+    /*
+     * Creates a new entry in the CMS for the Index Migration's progress.  Returns an Optional; if the document was
+     * created, it will be the created entry and empty otherwise.
+     */
+    public Optional<CmsEntry.Index> createIndexEntry();
+
+    /*
+     * Attempt to retrieve the Index Migration entry from the CMS, if it exists.  Returns an Optional; if the document
+     * exists, it will be the retrieved entry and empty otherwise.
+     */
+    public Optional<CmsEntry.Index> getIndexEntry();
+
+    /*
+     * Updates the Index Migration entry in the CMS.  Returns an Optional; if the document was updated,
+     * it will be the updated entry and empty otherwise.
+     */
+    public Optional<CmsEntry.Index> updateIndexEntry(CmsEntry.Index newEntry, CmsEntry.Index lastEntry);
+
+    /*
+     * Creates a new entry in the CMS for an Index Work Item.  Returns an Optional; if the document was
+     * created, it will be the created entry and empty otherwise.
+     */
+    public Optional<CmsEntry.IndexWorkItem> createIndexWorkItem(String name, int numShards);
```

Comment:
I'm not clear on the naming is this an index work item, seems like a index shard work item